src/include/duckdb/main/config.hpp

```diff
@@ -61,6 +61,37 @@ typedef void (*reset_global_function_t)(DatabaseInstance *db, DBConfig &config);
 typedef void (*reset_local_function_t)(ClientContext &context);
 typedef Value (*get_setting_function_t)(ClientContext &context);
 
+struct NumericSetting {
```

Comment:
Open to a better name
Basically I need to differentiate between a value set by a user and a setting set by us

Since the default settings are variable, there is no way to check after the fact if the current value was a default one or not

Reply:
ContextualizedSetting?
VariableDefaultSetting?